frontend/dockerfile/dockerfile2llb/convert.go

```diff
@@ -109,12 +110,51 @@ func Dockefile2Outline(ctx context.Context, dt []byte, opt ConvertOpt) (*outline
 	return &o, nil
 }
 
+func lintSourceInfo(srcMap *llb.SourceMap, srcRange *parser.Range) (filename string, source []string) {
+	if srcMap == nil {
+		return
+	}
+	filename = srcMap.Filename
+	if srcRange == nil {
+		return
+	}
+	lines := strings.Split(string(srcMap.Data), "\n")
+	start := srcRange.Start.Line
+	end := srcRange.End.Line
+	source = lines[start-1 : end]
+	return
+}
+
+func DockerfileLint(ctx context.Context, dt []byte, opt ConvertOpt) ([]linter.Warning, error) {
```

Comment:
This function belongs more to #4761 I think

Reply:
Agreed! Kept it in for testing purposes, but with the testing related comment below, going to refactor those, so this can get the boot from here!